tests/testthat/test-as-data-frame.R

```diff
@@ -77,15 +77,29 @@ with_mock_crunch({
             as.vector(ds$gender))
     })
 
+    test_that("coerceVariable produces expected list entries", {
+        df <- lapply(ds, as.vector)
+        names(df) <- names(ds)
+        df[] <- lapply(df, as.character)
+        df <- structure(df, class = "data.frame", row.names=c(NA, -nrow(ds)))
+        expect_is(coerceVariable(ds$birthyr, df)[[1]], "numeric")
+        expect_is(coerceVariable(ds$location, df)[[1]], "factor")
+        expect_is(coerceVariable(ds$textVar, df)[[1]], "character")
+        expect_is(coerceVariable(ds$starttime, df)[[1]], "Date")
+        expect_is(coerceVariable(ds$mymrset, df)[[1]], "factor")
+        expect_is(coerceVariable(ds$catarray, df)[[1]], "factor")
+    })
+
     test_that("as.data.frame when a variable has an apostrophe in its alias", {
         t2 <- ds
         t2@variables@index[[2]]$alias <- "Quote 'unquote' alias"
         expect_is(as.data.frame(t2), "CrunchDataFrame")
     })
 
     test_that("as.data.frame(as.data.frame())", {
-        expect_true(is.data.frame(as.data.frame(as.data.frame(ds))))
-        expect_true(is.data.frame(as.data.frame(ds, force=TRUE)))
```

Comment:
We should find a way to re-write these tests / update mocks so that they can still be run without the backend present.

Reply:
I just checked, and it looks like this would also bump our line coverage a lot. It looks like `as.data.frame.CrunchDataFrame` is not hit at all during travis's testing: https://codecov.io/gh/Crunch-io/rcrunch/pull/165/src/R/as-data-frame.R